### PR TITLE
Added windows support for static files

### PR DESCRIPTION
**Pull request by veitbjarsch**
## Changes
- Added windows support for static files

On windows machines it did not split up the path for static files into it's parts, because it was only searching for posix separators.

## Testing
Run the build on windows machines an check the _routes.json before and after.

### CODE REVIEW

**Comment by veitbjarsch** [comment]
@alexanderniebuhr fixed the formatting and linting error. PR should be good to go now.

**Comment by bluwy** [comment]
Do you have a repro or test for this? Checking the Astro source code, I don't think a backslash could happen in the first place. It'd be a bug there.

**Comment by veitbjarsch** [comment]
@bluwy Here is a minimal reproduction of the issue. Just installed asto (latest version) and the astro cloudflare plugin.
https://github.com/veitbjarsch/astro-cloudflare-demo

When I add folders in the public folder and run `npx astro build` on windows the reoutes.json looks like this:
https://github.com/veitbjarsch/astro-cloudflare-demo/blob/main/dist/_routes.json

**Comment by bluwy** [comment]
With your adjusted code, it makes more sense now and I think that's the right fix. Initially you made the change for Astro's `paths` which shouldn't have this issue.

File: packages/cloudflare/src/utils/generate-routes-json.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ export async function createRoutesFile(
 			const staticPath = staticFile;
 
 			const segments = removeLeadingForwardSlash(staticPath)
-				.split(posix.sep)
+				.split(process.platform === 'win32' ? win32.sep : posix.sep)
```
Comment (bluwy) [inline packages/cloudflare/src/utils/generate-routes-json.ts:189]: Could we simplify this as:

```js
import path from 'node:path'

//...
.split(path.sep)
```

Comment (veitbjarsch) [reply packages/cloudflare/src/utils/generate-routes-json.ts:189]: Yes looks like this will work too. I'll adjust it on Monday.

Comment (veitbjarsch) [reply packages/cloudflare/src/utils/generate-routes-json.ts]: @alexanderniebuhr @bluwy done.
Can you run the workflow tests once more.

**Review by bluwy — APPROVED** [review]

**Comment by alexanderniebuhr** [comment]
@veitbjarsch can you add an changeset with `pnpm exec changeset`

**Comment by veitbjarsch** [comment]
@alexanderniebuhr done

**Comment by alexanderniebuhr** [comment]
@veitbjarsch I can't see it, have you pushed your commit?

**Pull request merged**